Sources/Utility/StringMangling.swift

```diff
@@ -12,6 +12,32 @@
 /// of them are programming-related.
 extension String {
 
+    /// Returns a form of the string that is a valid bundle identifier
+    public func mangledToBundleIdentifier() -> String {
+        let mangledUnichars: [UnicodeScalar] = self.unicodeScalars.map {
+            switch $0.value {
+            case
+            // A-Z
+            0x0041...0x005A,
+            // a-z
+            0x0061...0x007A,
+            // 0-9
+            0x0030...0x0039,
+            // -
+            0xF2,
+            // .
+            0x2E:
+                return $0
+            default:
+                return "-"
+            }
+        }
+
+        // FIXME: There doesn't seem to be a way to create a string from an
+        //        array of Unicode scalars; but there must be a better way.
+        return mangledUnichars.reduce(""){ $0 + String($1) }
```

Comment:
There should be a fast way to create a string from a UTF-8 byte sequence, though, so it might be more efficient to do that here.  It might result in more `-` characters, though, but I don't know if that matters.